scaletest/prebuilds/run.go

```diff
@@ -0,0 +1,365 @@
+	defer cancel()
+
+	tkr := r.cfg.Clock.TickerFunc(workspacesCtx, workspacesPollInterval, func() error {
+		workspaces, err := r.client.Workspaces(workspacesCtx, codersdk.WorkspaceFilter{
+			Template: r.template.Name,
+		})
+		if err != nil {
+			return xerrors.Errorf("list workspaces: %w", err)
+		}
+
+		acquiredCount := 0
+		succeededCount := 0
+
+		for _, ws := range workspaces.Workspaces {
+			if ws.LatestBuild.Job.Status == codersdk.ProvisionerJobRunning ||
+				ws.LatestBuild.Job.Status == codersdk.ProvisionerJobSucceeded {
+				acquiredCount++
+			}
+			if ws.LatestBuild.Job.Status == codersdk.ProvisionerJobSucceeded {
+				succeededCount++
+			}
+		}
+
+		if r.prebuildJobCreationLatency == 0 && len(workspaces.Workspaces) >= targetNumWorkspaces {
+			// All jobs created
+			r.prebuildJobCreationLatency = time.Since(testStartTime)
+			r.cfg.Metrics.RecordJobCreation(r.prebuildJobCreationLatency, r.template.Name)
+		}
+
+		if r.prebuildJobAcquiredLatency == 0 && acquiredCount >= targetNumWorkspaces {
+			// All jobs acquired
+			r.prebuildJobAcquiredLatency = time.Since(testStartTime)
+			r.cfg.Metrics.RecordJobAcquired(r.prebuildJobAcquiredLatency, r.template.Name)
+		}
+
+		if succeededCount >= targetNumWorkspaces {
+			// All jobs succeeded
+			r.prebuildTotalLatency = time.Since(testStartTime)
+			r.cfg.Metrics.RecordCompletion(r.prebuildTotalLatency, r.template.Name)
+			return errTickerDone
+		}
+
+		return nil
+	}, "waitForPrebuildWorkspaces")
+	err := tkr.Wait()
+	if !xerrors.Is(err, errTickerDone) {
+		r.cfg.Metrics.AddError(r.template.Name, "wait_for_workspaces")
+		return xerrors.Errorf("wait for workspaces: %w", err)
+	}
+
+	logger.Info(ctx, "all prebuild workspaces created successfully", slog.F("template_name", r.template.Name), slog.F("duration", time.Since(testStartTime).String()))
+	return nil
+}
+
+func (r *Runner) measureDeletion(ctx context.Context, logger slog.Logger) error {
+	deletionStartTime := time.Now().UTC()
+	const deletionPollInterval = 500 * time.Millisecond
+
+	targetNumWorkspaces := r.cfg.NumPresets * r.cfg.NumPresetPrebuilds
+
+	deletionCtx, cancel := context.WithTimeout(ctx, r.cfg.PrebuildWorkspaceTimeout)
+	defer cancel()
+
+	tkr := r.cfg.Clock.TickerFunc(deletionCtx, deletionPollInterval, func() error {
+		workspaces, err := r.client.Workspaces(deletionCtx, codersdk.WorkspaceFilter{
+			Template: r.template.Name,
+		})
+		if err != nil {
+			return xerrors.Errorf("list workspaces: %w", err)
+		}
+
+		currentCount := len(workspaces.Workspaces)
+		deletingCount := 0
+
+		for _, ws := range workspaces.Workspaces {
+			if ws.LatestBuild.Transition == codersdk.WorkspaceTransitionDelete {
+				if ws.LatestBuild.Job.Status == codersdk.ProvisionerJobRunning ||
+					ws.LatestBuild.Job.Status == codersdk.ProvisionerJobSucceeded {
+					deletingCount++
+				}
+			}
+		}
+
+		if r.prebuildDeletionJobCreationLatency == 0 && (currentCount < targetNumWorkspaces || deletingCount > 0) {
```

Comment:
The metric is supposed to be when _all_ workspaces have their deleting job created, but this will trigger if _any_ deleting jobs are created.